### PR TITLE
[bitnami/kibana] Revisit Helm Chart tests

### PR DESCRIPTION
**Pull request by FraPazGal**
### Description of the change

After reviewing the existing tests and the general testing strategy we have been following until now, we saw there was a need to revisit them. The main objective that our chart testing strategy should aim for is asserting that the Chart deliverable works. As such, the focus should not be on the application itself or the container, which are regarded as trustful components (that is, they should have been tested at a previous stage).

As charts usually pack a number of different components, it is essential to test that their integrations and the application itself work as expected.

Consequently, we have updated the tests so they are related to the chart deliverable and left only a couple of basic tests to verify that the application functions correctly. Besides that, an effort has been made to reduce future tests' flakiness improving some existing selectors and checks.

Tested in https://github.com/FraPazGal/charts/pull/78

### Checklist

- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: .vib/kibana/cypress/cypress/integration/kibana_spec.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ before(() => {
   skipTheWelcomeScreen();
 });
 
-it('allows uploading a file', () => {
+it('allows uploading data', () => {
   cy.get('a[data-test-subj="uploadFile"]').click();
```
Comment (joancafom) [inline .vib/kibana/cypress/cypress/integration/kibana_spec.js:10]: These types of selectors tend to be pretty much accurate, is the element type really needed here?
```suggestion
  cy.get('[data-test-subj="uploadFile"]').click();
```

File: .vib/kibana/cypress/cypress/integration/kibana_spec.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ before(() => {
...
   cy.get('input#filePicker').selectFile(
```
Comment (joancafom) [inline .vib/kibana/cypress/cypress/integration/kibana_spec.js:11]: We usually use `[type="file"]`, which should be more resilient. I think you can also use it here.

File: .vib/kibana/cypress/cypress/integration/kibana_spec.js [hunk 3]
```diff
@@ -21,113 +21,44 @@ it('allows uploading a file', () => {
...
-  cy.fixture('dashboards').then((dashboard) => {
-    cy.contains(`${dashboard.newDashboard.title}.${random}`);
-  });
-});
-
-it('allows creating a visualisation', () => {
+it('allows creating a dashboard and visualisation', () => {
   cy.visit('app/visualize/');
   closeThePopups();
   cy.contains('button', 'Create').click();
```
Comment (joancafom) [inline .vib/kibana/cypress/cypress/integration/kibana_spec.js:33]: Use the `data-test-subj` attribute whenever possible.

File: .vib/kibana/cypress/cypress/integration/kibana_spec.js [hunk 4]
```diff
@@ -21,113 +21,44 @@ it('allows uploading a file', () => {
...
-  cy.get('[data-test-subj="visType-vega"]')
-    .should('contain.text', 'Custom visualization')
-    .forceClick();
+  cy.get('[data-test-subj="visType-vega"]').forceClick();
   cy.contains('[data-test-subj="visualizeSaveButton"]', 'Save').forceClick();
```
Comment (joancafom) [inline .vib/kibana/cypress/cypress/integration/kibana_spec.js:35]:
```suggestion
  cy.contains('[data-test-subj="visualizeSaveButton"]').forceClick();
```

File: .vib/kibana/cypress/cypress/integration/kibana_spec.js [hunk 5]
```diff
@@ -21,113 +21,44 @@ it('allows uploading a file', () => {
...
-    cy.get('[data-test-subj="dashboardDescription"]').type(
+    cy.get('[data-test-subj="viewDescription"]').type(
       `${visualization.newVisualization.description}.${random}`
     );
   });
+  cy.get('[id="new-dashboard-option"]').forceClick();
   cy.contains(
     '[data-test-subj="confirmSaveSavedObjectButton"]',
-    'Save'
+    'Save and go'
```
Comment (joancafom) [inline .vib/kibana/cypress/cypress/integration/kibana_spec.js:47]: The `data-test-subj` attribute should be specific enough.
```suggestion
```

File: .vib/kibana/cypress/cypress/integration/kibana_spec.js [hunk 6]
```diff
@@ -21,113 +21,44 @@ it('allows uploading a file', () => {
...
   ).forceClick();
-  cy.visit('/app/dashboards/list');
+  cy.contains('[data-test-subj="dashboardSaveMenuItem"]', 'Save').click();
```
Comment (joancafom) [inline .vib/kibana/cypress/cypress/integration/kibana_spec.js:49]:
```suggestion
  cy.contains('[data-test-subj="dashboardSaveMenuItem"]').click();
```

File: .vib/kibana/cypress/cypress/integration/kibana_spec.js [hunk 7]
```diff
@@ -21,113 +21,44 @@ it('allows uploading a file', () => {
...
-  cy.contains('span', 'Add element').forceClick({
-    multiple: true,
-  });
-  cy.contains('span', 'Text').forceClick();
-  cy.visit('app/canvas#/');
-  cy.fixture('workpads').then((workpad) => {
-    cy.contains(`${workpad.newWorkpad.name}${random}`);
+    cy.contains(
+      '[data-test-subj="confirmSaveSavedObjectButton"]',
+      'Save'
```
Comment (joancafom) [inline .vib/kibana/cypress/cypress/integration/kibana_spec.js:59]: Same here 

```suggestion
```

**Review by joancafom — COMMENTED** [review]
Thanks for your PR. I left you some comments, could you please review them?

File: .vib/kibana/cypress/cypress/integration/kibana_spec.js (hunk 1 above)
Comment (agomezmoron) [reply .vib/kibana/cypress/cypress/integration/kibana_spec.js:10]: Agree

**Review by agomezmoron — APPROVED** [review]
LGTM with @joancafom's feedback

File: .vib/kibana/cypress/cypress/integration/kibana_spec.js (hunk 4 above)
Comment (FraPazGal) [reply .vib/kibana/cypress/cypress/integration/kibana_spec.js]: Modified every affected check.

File: .vib/kibana/cypress/cypress/integration/kibana_spec.js (hunk 1 above)
Comment (FraPazGal) [reply .vib/kibana/cypress/cypress/integration/kibana_spec.js]: Modified every affected check. Thanks!

File: .vib/kibana/cypress/cypress/integration/kibana_spec.js (hunk 3 above)
Comment (FraPazGal) [reply .vib/kibana/cypress/cypress/integration/kibana_spec.js]: Modified every affected check.

**Review by joancafom — APPROVED** [review]
LGTM! Thanks for your changes 🚀

**Pull request merged**